### ISSUE TITLE
Add a way to automatically add frames with staircase pattern for opacity settings

### ISSUE THREAD

**Issue by pwab** [senshu/Sozi#480]
### Summary of your feature request
**For what purpose?**
We are working on online teaching material for students and Sozi is the only software able to 'convert' SVGs in presentation slides which can also be embedded into websites. We have a lot of SVGs which we would like to convert and they include methods of drawing lines and circles (geometric constructions) one after the other. So we are looking for a way to show each step one by one. We are using Sozis ability to change the opacity of layers for every frame. This works but we think it could be automated.

**How to do it now**
In Sozi 20.03 it is now possible to add all layers at once. Those layers are sorted in the same order in which they appear in the SVG/Inkscape. If you want to create a Sozi file where each layer is shown one by one (each frame unhides the next layer) you have to do the following (systematic approach):

1. Add all layers
2. Add a frame for each layer
3. Hide all layers in every frame by changing the opacity to 0
4. Unhide the layers one by one in the corresponding frame and the frames that follow

You will get a **staircase pattern** for a file where the layers are sorted from bottom to top because in every frame a new layer is shown and so the opacity does. Just a little sample:

**3 layers, 3 frames, opacity shown in % as 0 or 100**
```
layers		frame_1		frame_2		frame_3
layer_3		0			0			100
layer_2		0			100			100
layer_1		100			100			100
```

**How it could be done in the future*
The aim of this feature request is to simplify this. It could be easliy automated with the click of a button so that for every layer a frame is added and then the opacity changes are set automatically.

### Version of Sozi and other relevant software

* OS: Windows 10 x64
* Inkscape: 0.92.4
* Sozi: 20.03

### Steps to reproduce the problem

See the description in the first paragraph.

### Expected behavior

**Gif of what it should look like in the end:**
![layer_after_layer](https://user-images.githubusercontent.com/19271185/80572495-ac155880-89fe-11ea-810b-0f025731837a.gif)

**Sample project:**
[layer_after_layer_sample.zip](https://github.com/senshu/Sozi/files/4550339/layer_after_layer_sample.zip)

_(Just in case you are wondering: Almost the same project sample is used in #479. This has nothing to do with the proposal here. I only stumbled upon the preview issue but the functionality still works on export so I'm using it here again.)_

### Observed behavior

It works now but we think it could be automated.

**Comment by senshu** [comment]
Thanks for submitting this feature request.
This is a very interesting idea and I think it could benefit many users.

While the use case is clear, maybe there are several ways to provide this feature in the user interface of Sozi.
It should be easy to add the staircase pattern after creating frames, or after reordering frames.

Currently, my preference would be to let the user select a set of layers and frames, and click on a "staircase" button.

**Comment by pwab** [comment]
> Currently, my preference would be to let the user select a set of layers and frames, and click on a "staircase" button.

That would be a very useful implementation because we got to some situations where we added extra steps in between two staircase patterns (like hiding already shown layers that would otherwise confuse).

**Comment by pwab** [comment]
Just a side question: Is it possible to hide elements such as rects, circles, lines or groups instead of a whole layer?

**Comment by senshu** [comment]
> Is it possible to hide elements such as rects, circles, lines or groups instead of a whole layer?

No. Sozi is not designed to work on specific elements of a presentation.
You can hide part of a layer using the "clip" feature, but it is not as flexible.
